pkg/sink/heroku.go

```diff
@@ -0,0 +1,61 @@
+package sink
+
+import (
+	"context"
+	"fmt"
+
+	heroku "github.com/heroku/heroku-go/v5"
+	"github.com/pkg/errors"
+)
+
+type HerokuServiceIface interface {
+	ConfigVarUpdate(ctx context.Context, appIdentity string, o map[string]*string) (heroku.ConfigVarUpdateResult, error)
+	ConfigVarInfoForApp(ctx context.Context, appIdentity string) (heroku.ConfigVarInfoForAppResult, error)
+}
+
+type HerokuSink struct {
+	BaseSink    `yaml:",inline"`
+	Client      HerokuServiceIface `yaml:"client"`
+	AppIdentity string             `yaml:"AppIdentity"`
+}
+
+func (sink *HerokuSink) WithHerokuClient(client HerokuServiceIface) *HerokuSink {
+	sink.Client = client
+	return sink
+}
+
+func (sink *HerokuSink) GetKeyToName() map[string]string {
+	return sink.KeyToName
+}
+
+func (sink *HerokuSink) WithKeyToName(m map[string]string) *HerokuSink {
+	sink.BaseSink = BaseSink{KeyToName: m}
+	return sink
+}
```

Comment:
These should probably be part of the `BaseSink` struct so they can be re=used by all the sinks. This is possible since we're embedding the BaseSink to all the other sinks such as https://github.com/chanzuckerberg/rotator/pull/56/files#diff-bb353a126a4e32697e377d488b70bcadR17